src/Symfony/Component/DependencyInjection/Definition.php

```diff
@@ -796,7 +800,7 @@ public function isDeprecated()
      */
     public function getDeprecationMessage($id)
     {
-        return str_replace('%service_id%', $id, $this->deprecationTemplate);
+        return null !== $this->deprecationTemplate ? str_replace('%service_id%', $id, $this->deprecationTemplate) : null;
```

Comment:
Maybe it would be better to use const?
```
const DEFAULT_DEPRECATION_TEMPLATE = 'The "%service_id%" service is deprecated. You should stop using it, as it will soon be removed.'
```